packages/components/src/font-size-picker/types.ts

```diff
@@ -81,18 +76,25 @@ export type FontSize = {
 	slug: string;
 };
 
-export type FontSizeOption = Omit< FontSize, 'size' > &
-	Partial< Pick< FontSize, 'size' > >;
-
-export type FontSizeSelectOption = Pick< FontSizeOption, 'size' > & {
-	key: string;
-	name?: string;
-	__experimentalHint: ReactNode;
+export type FontSizePickerSelectProps = {
+	fontSizes: FontSize[];
+	value: FontSizePickerProps[ 'value' ];
+	size: FontSizePickerProps[ 'size' ];
+	onChange: ( value: number | string | undefined ) => void;
+	onSelectCustom: () => void;
 };
 
-export type FontSizeToggleGroupOption = {
+export type FontSizePickerSelectOption = {
 	key: string;
-	value: number | string;
-	label: string;
 	name: string;
+	value?: FontSize[ 'size' ];
+	__experimentalHint?: string;
+};
+
+export type FontSizePickerToggleGroupProps = {
+	fontSizes: FontSize[];
+	value: FontSizePickerProps[ 'value' ];
+	__nextHasNoMarginBottom?: boolean;
+	size: FontSizePickerProps[ 'size' ];
+	onChange: ( value: number | string | undefined ) => void;
```

Comment:
Same as previous comment

```suggestion
export type FontSizePickerToggleGroupProps = Pick<
	FontSizePickerProps,
	'value' | 'size' | '__nextHasNoMarginBottom'
> & {
	fontSizes: NonNullable< FontSizePickerProps[ 'fontSizes' ] >;
	onChange: NonNullable< FontSizePickerProps[ 'onChange' ] >;
```